docs(readme): added description of the tool (#9)
input-output-hk/Lean-blaster: README.md

In order to use Blaster, your project needs to depend on `lean-blaster`. 

### Using lakefile.toml

> In you use `lakefile.toml`, simply add a dependency to this repository:

Reviewer: ```suggestion
If you use `lakefile.toml` then, simply add a dependency to this repository:
```

```toml
[[require]]
name = "Solver"
git = "https://github.com/input-output-hk/Lean-blaster"
rev = "main"
```

### Using lakefile.lean

> If you use `lakefile.lean`, simply add a dependency to this repository:

Reviewer: ```suggestion
If you use `lakefile.lean` then, simply add a dependency to this repository:
```

```lean4
require «Solver» from git
  "https://github.com/input-output-hk/Lean-blaster" @ "main"
```

### Solver options

>   - `unfold-depth`: specifying the number of unfolding to be performed on recursive functions (default: 100)

Author: ```suggestion
```

  - And many more.

- **Match Expression Optimization**: `match` expressions are optimized by:
  - **Constant Propagation**: If a discriminator (the value being matched on) is a known constant, the `match` expression is replaced with the corresponding branch.
  - **Unreachable Branch Elimination**: If a branch is determined to be unreachable based on the current context, it is eliminated.
  - **Normalization**: In some cases, `match` expressions are normalized into a series of `if-then-else` expressions to enable further simplification.

#### Function propagation

Function propagation is another key optimization strategy, detailed in `Solver/Optimize/Rewriting/FunPropagation.lean`. This technique simplifies expressions by "pushing" function calls into their arguments. For example, a function applied to an `if-then-else` expression can be transformed into an `if-then-else` expression where the function is applied to both the `then` and `else` branches. This can be particularly effective when one of the branches can be further simplified after the function is applied using other optimization rules.

### Second-step: SMT Translation

> Using the whole set of optimization rules, it happens that the theorem being proved will reduce to `True`. This concludes the proof, and the theorem is `Valid`. In some instances, it will happen that it is reduced to `False` and the theorem is therefore `Falsified`. Most of the time, the theorem expression will not have been reduced to either and the solver will then translate the remaining Lean expression into the SMT-LIB format.

Reviewer: ```suggestion
Using the whole set of optimization rules, it may happen that a theorem can be reduced to `True`.  This concludes the proof and the theorem is considered as `Valid`. For some other cases, a theorem may also be reduced to `False` and will therefore be declared as `Falsified`. Most of the time, a proof might not be concluded at the optimization phase. In this case, the optimized Lean expression is translated into an SMT-LIB format and submitted to the backend solver.
```

> This is handled in `Solver/Smt/Translate.lean`. This process involves several key steps:

Reviewer: ```suggestion
The translation step is handled in `Solver/Smt/Translate.lean`. This process involves several key steps:
```
